### PR TITLE
Allow for TMDB trailer collection

### PR DESCRIPTION
**Pull request by jimdogx**
Having recently gained access to a server with a TMDB Trailer folder, I found that "Trailers" were not a supported collection type.  This small change enables the ability to view them. 

**Changes**
* Support TMDB Trailers.
* Add default "Filter" option as that could cause a crash if the user went in to Filter options with nothing defined.

### CODE REVIEW

**Comment by cewert** [comment]
Are you talking about this jellyfin plugin? How do I test this? https://github.com/crobibero/jellyfin-plugin-tmdb-trailers

**Comment by jimdogx** [comment]
> Are you talking about this jellyfin plugin? How do I test this? https://github.com/crobibero/jellyfin-plugin-tmdb-trailers

Sorry, I should have included that information.

Yes, it is using that plugin.  The current settings for the plugin are: 

- Enable Movie Trailers Channel
- Enable Upcoming Movies Trailers
- Enable Now Playing Moves Trailers

But of course you can test with other settings.

**Comment by cewert** [comment]
Sweet, thanks. This plugin looks amazing I'm excited to get it set up. I've never had trailer support setup.

I thought we didn't support plugins though? Or how do you look that up which ones are 3rd party or whatever

**Review by cewert — CHANGES_REQUESTED** [review]
Here's my testing notes. I don't have any TV channels setup so I'm not sure I can test the "Enable Movie Trailers Channel" part

- The thumbnail images for the trailers on the homerow didn't look good. like the resolution is good but its zoomed in instead of resized for most of them.
- When selecting "ok" on a trailer from the home screen I got a popup that said type: trailer wasn't supported
- When selecting "play" on a trailer on the homerow, no error popup but no video either
- The "tmdb trailors" collection on my the top row of the home screen correctly displays and let's me play the trailers
- The quick play feature doesn't work when viewing the trailers using the itemgrid
- EDIT: 99% of the trailers were missing metadata. The 3 points below are for the 99%. One trailer pulled in meta data and had no problems on the Roku, M3GAN
  - When viewing a trailer, there is no description. This is probably what we are given from the plugin but it sucks there is no release date or anything
  - When viewing a trailer, the "extras" dark background overlay is still shown even though there is nothing to display
  - When viewing a trailer, after hitting "down" to view the extras panel, you are unable to hit "up" to get rid of the panel again and must hit "back" gain control of the screen again and are sent to the trailer itemgrid



I think we need to fix the home screen stuff and the buggy extras panel when viewing the trailer before we merge. The rest can wait if you want

**Comment by cewert** [comment]
I have no idea how the plugin works but somehow the web client pulls in some good metadata on the trailers. If possible, we should try to grab the same... description, release year, genre, community rating, rotten tomatoes rating, and film rating (R, PG-13 etc.) Picture for reference:
tmdb-trailers-webclient

EDIT: This is the only trailer that pulled in metadata for me. Both on the web client and on roku

**Comment by 1hitsong** [comment]
- [ ] Collection images under My Media are missing. It's use the TV fallback icon - web client has images
- [ ] Same as above for TMDb Extras. No images, web client has images
- [ ] I added a trailer to my favorites. It shows and plays on the homepage Favorites section, but it has no image

File: components/ItemGrid/LoadItemsTask2.brs [hunk 1]
```diff
@@ -226,7 +226,8 @@ sub loadItems()
                 tmp.json = item
                 tmp.type = "Folder"
                 tmp.posterUrl = api_API().items.getimageurl(item.id, "primary", 0, { "maxHeight": 280, "maxWidth": 280, "quality": "90" })
-
+            else if item.Type = "Trailer"
```
Comment (1hitsong) [inline components/ItemGrid/LoadItemsTask2.brs:236]: All string comparisons should force case.
```suggestion
            else if LCase(item.Type) = "trailer"
```

File: components/home/HomeItem.brs [hunk 2]
```diff
@@ -231,6 +231,12 @@ sub itemContentChanged()
         return
     end if
 
+    if itemData.type = "Trailer"
```
Comment (1hitsong) [inline components/home/HomeItem.brs:258]: All string comparisons should force case.
```suggestion
    if LCase(itemData.type) = "trailer"
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Comment by cewert** [comment]
Checked all the boxes in the plugin's settings and now I have like 80 trailers :smile:

Anyway, I noticed I have a trailer with partial metadata, just release year. And also the web client somehow made the "more like this" row even though the genre is missing. Another thing I noticed is that trailers are showing up under "more like this". That's pretty cool.

tmdb-trailers-webclient-partial-metadata

**Comment by cewert** [comment]
I know this doesn't say merge conflict but the required workflow won't run until you merge unstable into your PR branch FYI

**Comment by jimdogx** [comment]
Closing this PR until I have some time to implement it properly and not just "hey I can play trailers now" which made me happy 😆

**Pull request closed without merging**